### PR TITLE
Use Docusaurus + Add Articles

### PR DESCRIPTION
**Pull request by thebrandonlucas**
This is a massive PR change to completely restructure the site to use [docusaurus](https://docusaurus.io/), a docs-site generator with all sorts of built-in bells and whistles that is commonly used and familiar. It allows us to write our articles in markdown and handles routing/rendering automatically. I kept the home page as mostly the same as our current payjoin.org site with some enhancements (and removed a couple other things). I've also been using excalidraw to create explainer diagrams for payjoin v1/v2. You can see them all [here](https://excalidraw.com/#json=5otIl9VukIxNML3D8uTpr,hGFROSW-lb5YuyNfY4SRtg).

This PR is not yet complete as I think we should write a couple of tutorials on how to use it (especially payjoin v2) as well as write some articles on building a send/receive payjoin. We can also discuss whether we want to port over PDK to this website and make it it's own dedicated page with tutorials and navigation (it can have it's own homepage on the same site, etc.) or whether we think it's better to keep them separate. I personally favor integrating everything we can for ease of access unless we feel the site is getting hard to navigate, but generally we should try to funnel people from very high level -> very technical in a naturally progressive way, which this docs site is designed to handle well. PDK website doesn't have much on it currently that would require porting, and I notice the actual API docs are outsourced to the rust documentation anyway, so we wouldn't be losing much in the port and would probably gain navigability.

We can also consider just putting the blog (or _a_ blog) on payjoin updates on here, as well as podcast appearances, news, video tutorials, etc. Docusaurus comes with a sample blog built-in for reference, I think that would fit well here.

This took a lot of inspiration from [fedimint's setup](https://fedimint.org/docs/intro). I like a lot about how that project is setup, in addition, their [Github](https://github.com/fedimint/fedimint) is also very well organized.

Overall, this change will make it much easier for us to iterate on documentation and educational materials. Let me know what you think.

Screenshot 2024-05-13 at 9 41 48 AM
Screenshot 2024-05-13 at 9 42 17 AM

Screenshot 2024-05-13 at 9 42 40 AM
Screenshot 2024-05-13 at 9 43 44 AM
Screenshot 2024-05-13 at 9 42 59 AM
Screenshot 2024-05-13 at 9 43 23 AM
Screenshot 2024-05-13 at 9 43 33 AM

### CODE REVIEW

**Comment by thebrandonlucas** [comment]
After discussion with @DanGould, it was agreed to revert any copy changes and and not to introduce any new content changes, so that this PR can instead focus on underlying structural changes to docusaurus to the extent possible, to reduce the amount of changes being introduced in one PR. However, since this is a major structural change to the underlying website code, this having many changes at once is inevitable to an extent.

To that end, I kept the content nearly exactly the same as the current website. The homepage remains mostly the same, with the following changes:
- The navbar is a bit different now due to docusaurus default styling. All the links are the same with the addition of a link to our GitHub page.
- Some slight changes to the copy style (using defaults)
- New (default docusaurus) footer, content kept the same
- Removed supported/unsupported wallets table (we can re-add later if we choose to do so, but should probably find a way to keep it in sync with the Payjoin wiki when/if we decide to redo it)
- Added redirects for the three routes we had at `/lightning`, `/scale`, `/privacy` for backward-compatibility
- The `/lightning`, `/scale`, `/privacy` articles as well as the FAQ are now located at `/docs`, which is what you get when you click "Learn" in the navbar.

Overall, I think this move will allow us to focus on the bigger picture and iterate on content much faster and not get stuck in minutiae of styling & design decisions. Docusaurus still allows a great deal of flexibility if we choose to do that.

I have tested all the links myself and they look good, let me know what you think!

File: blog/2019-05-29-long-blog-post.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+tags: [hello, docusaurus]
+---
+
+This is the summary of a very long blog post,
+
+Use a `<!--` `truncate` `-->` comment to limit blog post size in the list view.
+
+<!--truncate-->
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque elementum dignissim ultricies. Fusce rhoncus ipsum tempor eros aliquam consequat. Lorem ipsum dolor sit amet
```
Comment (DanGould) [inline blog/2019-05-29-long-blog-post.md:14]: I find it odd to commit Lorem ipsum

File: blog/2019-05-28-first-blog-post.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+title: First Blog Post
+authors:
+  name: Gao Wei
+  title: Docusaurus Core Team
+  url: https://github.com/wgao19
+  image_url: https://github.com/wgao19.png
+tags: [hola, docusaurus]
+---
+
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque elementum dignissim ultricies. Fusce rhoncus ipsum tempor eros aliquam consequat. Lorem ipsum dolor sit amet
```
Comment (DanGould) [inline blog/2019-05-28-first-blog-post.md:12]: Lorem ipsum?

File: blog/2021-08-01-mdx-blog-post.mdx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+slug: mdx-blog-post
+title: MDX Blog Post
+authors: [slorber]
+tags: [docusaurus]
+---
+
+Blog posts support [Docusaurus Markdown features](https://docusaurus.io/docs/markdown-features), such as [MDX](https://mdxjs.com/).
```
Comment (DanGould) [inline blog/2021-08-01-mdx-blog-post.mdx:8]: we seem not to render this post

File: blog/2021-08-26-welcome/docusaurus-plushie-banner.jpeg
Comment (DanGould) [inline blog/2021-08-26-welcome/docusaurus-plushie-banner.jpeg:1]: We are not using this file, are we?

File: blog/2021-08-26-welcome/index.md
Comment (DanGould) [inline blog/2021-08-26-welcome/index.md:1]: unused

File: blog/authors.yml
Comment (DanGould) [inline blog/authors.yml:1]: unused

File: src/components/HomepageFeatures/styles.module.css
Comment (DanGould) [inline src/components/HomepageFeatures/styles.module.css:1]: Is this file used?

File: static/img/docusaurus-social-card.jpg
Comment (DanGould) [inline static/img/docusaurus-social-card.jpg:1]: I think when the site is linked this is going to show up. We may not want that

File: static/img/favicon.ico
Comment (DanGould) [inline static/img/favicon.ico:1]: did the old favicon get replaced?

File: static/img/undraw_docusaurus_mountain.svg
Comment (DanGould) [inline static/img/undraw_docusaurus_mountain.svg:1]: Are these docusaurus images used?

File: README.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
 
-```
+Official website for the Payjoin protocol.
+
+To run locally:
+```shell
 yarn
-yarn run dev
-```
+yarn start
```
Comment (DanGould) [inline README.md:8]: I suppose one must run `yarn install` first

**Review by DanGould — CHANGES_REQUESTED** [review]
Overall this is exactly what I was looking for. There are just a few files (which I believe I left comments on) which are part of the template that will never be used in this project which I would like to see removed

File: README.md (hunk 4 above)
Comment (thebrandonlucas) [reply README.md:8]: `yarn` is a shorthand for `yarn install` and has the same effect. I changed it to `yarn install` to clear up confusion.

File: blog/2019-05-28-first-blog-post.md (hunk 2 above)
Comment (thebrandonlucas) [reply blog/2019-05-28-first-blog-post.md]: Oversight on my part, these template blog posts I originally intended to modify into our substack posts and came with the template docusaurus. We can add this later if we feel the need to.

File: .prettierignore [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
+.docusaurus
 .env
 .env.*
 !.env.example
 
-# Ignore files for PNPM, NPM and YARN
-pnpm-lock.yaml
-package-lock.json
+# Ignore files for YARN
 yarn.lock
```
Comment (DanGould) [inline .prettierignore:11]: lock files ensure my build and your build use the same dependencies in different environments. AFAIK they are usually only ignored when developing libraries. Any reason to ignore here?

File: README.md (hunk 4 above)
Comment (DanGould) [reply README.md:8]: TIL!

File: src/components/UsedBy/index.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      </Marquee>
+      {/* <div>
+        <a href="https://payjoindevkit.org/" target="_blank">
+          <button className="font-extrabold  bg-gradient-to-r bg-secondary border-white border-2 border-solid shadow-[0px_0px_10px_10px_rgba(0,0,0,0.3)] shadow-primary hover:scale-105 transition-all p-8 rounded-lg cursor-pointer text-5xl ">
+            <span className="text-transparent bg-clip-text bg-gradient-to-r from-primary to-white">
+              Integrate Payjoin
+            </span>
+          </button>
+        </a>
+      </div> */}
```
Comment (thebrandonlucas) [inline src/components/UsedBy/index.tsx:62]: I removed this fancy "Integrate Payjoin" button for the time being in favor of maintaining parity with the current site.

File: static/img/docusaurus-social-card.jpg
Comment (DanGould) [inline static/img/docusaurus-social-card.jpg:1]: hopefully we find some place to use this

File: static/img/docusaurus.png
Comment (DanGould) [inline static/img/docusaurus.png:1]: is this used?

File: static/img/logo.svg
Comment (DanGould) [inline static/img/logo.svg:1]: Is this used?

File: static/img/undraw_docusaurus_mountain.svg
Comment (DanGould) [inline static/img/undraw_docusaurus_mountain.svg:1]: unused?

File: static/img/undraw_docusaurus_react.svg
Comment (DanGould) [inline static/img/undraw_docusaurus_react.svg:1]: unused?

File: static/img/undraw_docusaurus_tree.svg
Comment (DanGould) [inline static/img/undraw_docusaurus_tree.svg:1]: Unused?

**Review by DanGould — CHANGES_REQUESTED** [review]
Some more unused files

File: .prettierignore (hunk 5 above)
Comment (thebrandonlucas) [reply .prettierignore:11]: This is `.prettierignore`, not `.gitignore`. `yarn.lock` is being committed, just not linted.

**Review by DanGould — APPROVED** [review]

**Comment by DanGould** [comment]
It's live on payjoin.org and exceeds my expectation on mobile! I'm so stoked you pulled this one through. 🚀 The new arch is gonna 10x what we can do with the website BOOM.

**Pull request merged**